registry/registry.go

```diff
@@ -61,6 +65,54 @@ func hasFile(files []os.FileInfo, name string) bool {
 	return false
 }
 
+// ReadCertsDirectory reads the directory for TLS certificates
+// including roots and certificate pairs and updates the
+// provided TLS configuration.
+func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
+	fs, err := ioutil.ReadDir(directory)
```

Comment:
It does appear that the rest of the file info is not even used. `Walk` is not ideal since the current logic makes an extra check within the list to see if both files for a key pair exist. More efficient would just be using the underlying directory file...

```
	d, err := os.Open(directory)
	if err != nil {
		return err
	}
	fs, err := d.Readdirnames(-1)
	d.Close()
	if err != nil {
		return err
	}
```